### ISSUE TITLE
Warning: .then() only accepts functions but was passed: [object Object]

### ISSUE THREAD

**Issue by joerx** [timkelty/shipit-shared#41]
Reproduce by simply running the tests. (node v4.4.2, node v5.9.1, bluebird@3.4.0) Generates the following output: 

```
Warning: .then() only accepts functions but was passed: [object Object]
    at /Users/jhenning/Devel/Contrib/shipit-shared/tasks/shared/set-permissions.js:36:8
From previous event:
    at Shipit.task (/Users/jhenning/Devel/Contrib/shipit-shared/tasks/shared/set-permissions.js:32:25)
    at module.exports (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/lib/runTask.js:34:7)
    at Shipit.Orchestrator._runTask (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/index.js:273:3)
    at Shipit.Orchestrator._runStep (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/index.js:214:10)
    at /Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/index.js:279:18
    at finish (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/lib/runTask.js:21:8)
    at /Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/lib/runTask.js:43:4
From previous event:
    at module.exports (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/lib/runTask.js:42:5)
    at Shipit.Orchestrator._runTask (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/index.js:273:3)
    at Shipit.Orchestrator._runStep (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/index.js:214:10)
    at Shipit.Orchestrator.start (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/orchestrator/index.js:134:8)
    at Context.<anonymous> (/Users/jhenning/Devel/Contrib/shipit-shared/test/test.js:219:14)
    at callFnAsync (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runnable.js:349:8)
    at Test.Runnable.run (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runnable.js:301:7)
    at Runner.runTest (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runner.js:422:10)
    at /Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runner.js:528:12
    at next (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runner.js:342:14)
    at /Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runner.js:352:7
    at next (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runner.js:284:14)
    at /Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runner.js:315:7
    at done (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runnable.js:287:5)
    at callFn (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runnable.js:344:7)
    at Hook.Runnable.run (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runnable.js:319:7)
    at next (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runner.js:298:10)
    at Immediate._onImmediate (/Users/jhenning/Devel/Contrib/shipit-shared/node_modules/mocha/lib/runner.js:320:5)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

Similar problem occurs in `create-dirs.js:64:8`.

Tracked it to the following snippet:

``` js
      return mapPromise(shipit.config.shared.dirs, setPermissions)
      .then(mapPromise(shipit.config.shared.files, setPermissions))
      .then(function() {
        shipit.log(chalk.green(util.format('Permissions set on %s.', shipit.config.shared.shipitMethod)));
      })
      .then(function() {
        shipit.emit('sharedPermissionsSet');
      });
```

`mapPromise()` returns a Promise object, not a function. Simple fix would be to wrap the argument for the first `.then()` into a function. Will submit a PR shortly.

**Comment by dbernar1** [comment]
Thanks @joerx, that helped.

**Comment by timkelty** [comment]
Fixed in 4.4.2, thanks @joerx!
